module/fallbackstrategy.go

```diff
@@ -0,0 +1,11 @@
+package module
+
+// FallbackStrategy defines an interface to handle usage of multiple
+// clients connected to different hosts.
+type FallbackStrategy interface {
+	// ClientIndex return index of client to use
+	ClientIndex() int
+
+	// Failure indicate failure to use client at active index
+	Failure()
+}
```

Comment:
A `Success(int)` method would be useful in general, and I think is required to fully reproduce the "fallback-after-2-consecutive-failures" logic we are trying to encapsulate in this component (otherwise we don't know when failures are consecutive).